### PR TITLE
Deprecate experimental testing utilities

### PR DESCRIPTION
**Pull request by jerelmiller**
Closes https://github.com/apollographql/apollo-client/issues/11929

Deprecates experimental testing utilities introduced in 3.10 in favor of [`graphql-testing-library`](https://github.com/apollographql/graphql-testing-library).

### CODE REVIEW

**Comment by jerelmiller** [comment]
@alessbell would love your thoughts here. Also, what would we like to do with the [schema driven testing doc](https://www.apollographql.com/docs/react/development-testing/schema-driven-testing)? Do we want to add a note to the top of that with a similar deprecation warning? Or should we rewrite that for 3.11 to switch over to graphql-testing-library?

File: src/testing/experimental/createTestSchema.ts [hunk 1]
```diff
@@ -50,6 +50,8 @@ interface TestSchemaOptions {
  * ```
  * @since 3.9.0
  * @alpha
+ * @deprecated `createTestSchema` is deprecated and will be removed in 3.12.0.
+ * Please migrate to [`graphql-testing-library`](https://github.com/apollographql/graphql-testing-library).
```
Comment (alessbell) [inline src/testing/experimental/createTestSchema.ts:54]: Not sure if we want the repo name here or the npm package name which is `@apollo/graphql-testing-library`. I have a slight preference for the latter but either works :)

**Review by alessbell — APPROVED** [review]
Thanks so much for opening this! Left one minor comment but this LGTM 🚀

File: src/testing/experimental/createTestSchema.ts (hunk 1 above)
Comment (jerelmiller) [reply src/testing/experimental/createTestSchema.ts]: Good callout! I agree, the package name seems more reasonable. I've updated it with [7b6c071](https://github.com/apollographql/apollo-client/pull/11930/commits/7b6c07183daadf518424f0f521ca388b5f2fb73c). Thanks!

**Pull request merged**